plugins/modules/apt_rpm.py

```diff
@@ -164,16 +203,32 @@ def main():
 
     p = module.params
 
+    out = ""
     if p['update_cache']:
-        update_package_db(module)
+        out += update_package_db(module)
 
+    if p['clean']:
+        out += clean(module)
+
+    if p['dist_upgrade']:
+        out += dist_upgrade(module)
+
+    # If package is missing return exit_json
     packages = p['package']
 
-    if p['state'] in ['installed', 'present']:
-        install_packages(module, packages)
+    if not isinstance(packages, list):
+        if out == '':
+            module.fail_json(msg="Missing command")
+
+        else:
+            module.exit_json(changed=True, msg=out)
```

Comment:
Why is `out` returned as `msg` here, but not in any other `module.exit_json` call in this module? That doesn't make sense from my POV.

Also why not return this as `stdout`?

Reply:
Because out has compound output of one or more command. Fox example, update_cache+dist_upgrade+clean.

Reply:
Yes, but why isn't that put into `stdout`? That's the standard return value for program output. (https://docs.ansible.com/ansible/latest/reference_appendices/common_return_values.html#stdout)